src/libraries/Native/Unix/System.Native/pal_networking.h

```diff
@@ -301,8 +301,16 @@ typedef struct
     uint32_t Padding;    // Pad out to 8-byte alignment
 } SocketEvent;
 
+PALEXPORT int32_t SystemNative_PlatformSupportsGetAddrInfoAsync(void);
+
 PALEXPORT int32_t SystemNative_GetHostEntryForName(const uint8_t* address, int32_t addressFamily, HostEntry* entry);
```

Comment:
New public exports will need to be added to the table in  entrypoints.c , so that they can be called in single-file configuration as well - when this is not a separate .so/.dylib.

Just need to add method names to the dispatch table.

There is a build time check that will get upset if the dispatch table and exports get out of sync.

Reply:
Thanks for the hints!

Just saw your https://github.com/dotnet/runtime/pull/45995 -- is this in the same vein?